### ISSUE TITLE
current keybase cli cannot login on old accounts!

### ISSUE THREAD

**Issue by iMartyn** [keybase/client#4440]
I have had to wipe my linux machine and have reimported my keys in gpg from a backup.

gpg is happy : 
`ranyardm@osxreally-af58 ~/Downloads $ gpg --list-keys`
`/home/ranyardm/.gnupg/pubring.gpg`
`---------------------------------`
`pub   2048R/CFAEF3F4 2014-01-10`
`uid                  Martyn Ranyard (2014 onwards) <martyn@ranyard.info>`
`uid                  keybase.io/imartyn <imartyn@keybase.io>`
`sub   2048R/5DC2D226 2014-01-10`

but keybase refuses to login :

`ranyardm@osxreally-af58 ~ $ keybase login`
`Your keybase username or email address: iMartyn`
`▶ ERROR Sorry, your account is already established with a PGP public key, but this`
`utility cannot find the corresponding private key on this machine.`
`This is the fingerprint of the PGP key in your account:`
` `
`   4C31 0155 7BA5 83F4 7FF7 9776 AB4A 6048 CFAE F3F4`
` `
`You need to prove you're you. We suggest one of the following:`
` `
`   - put one of the PGP private keys listed above on this machine and try again`
`   - reset your account and start fresh: https://keybase.io/#account-reset`

Please don't make me revoke my pgp key and all my proofs :(

**Comment by maxtaco** [comment]
Run `gpg -K`?

**Comment by iMartyn** [comment]
`ranyardm@osxreally-af58 ~/src/grafana_ng (BPE-49) $ gpg -K`
`/home/ranyardm/.gnupg/secring.gpg`
`---------------------------------`
`sec   2048R/CFAEF3F4 2014-01-10`
`uid                  Martyn Ranyard (2014 onwards) <martyn@ranyard.info>`
`uid                  keybase.io/imartyn <imartyn@keybase.io>`
`ssb   2048R/5DC2D226 2014-01-10`

**Comment by maxtaco** [comment]
Can you please run a `keybase log send`?

**Comment by iMartyn** [comment]
Log id 5acb5891d19f932462283a1c

**Comment by iMartyn** [comment]
`▶ WARNING error opening log "/home/ranyardm/.cache/keybase/keybase.kbfs.log": open /home/ranyardm/.cache/keybase/keybase.kbfs.log: no such file or directory`
`▶ WARNING error opening log "/home/ranyardm/.cache/keybase/Keybase.app.log": open /home/ranyardm/.cache/keybase/Keybase.app.log: no such file or directory`
`▶ WARNING error opening log "/home/ranyardm/.cache/keybase/keybase.updater.log": open /home/ranyardm/.cache/keybase/keybase.updater.log: no such file or directory`
`▶ WARNING error opening log "/home/ranyardm/.cache/keybase/keybase.start.log": open /home/ranyardm/.cache/keybase/keybase.start.log: no such file or directory`
 whilst running that btw.

**Comment by iMartyn** [comment]
Also fwiw, `rm -rf ~/.config/keybase ~/.cache/keybase` makes no difference either :(

**Comment by iMartyn** [comment]
Also, debian wheezy ships with gpg1 as `gpg`, and gpg2 as `gpg2`, both see my keys correctly, and swapping them over makes no difference (`/usr/bin/gpg` as a symlink to `/usr/bin/gpg2`)

**Comment by patrickxb** [comment]
Those WARNING messages are fine.

I'm looking at your log right now.  I see the output of `gpg -K` above, but could you also run

    gpg -K --with-colons --fingerprint

I'm guessing we have a bug parsing the output of that as we can't seem to find CFAEF3F4.

**Comment by iMartyn** [comment]
there's interesting... that returns no output!

**Comment by patrickxb** [comment]
how about

    gpg2 -K --with-colons --fingerprint

**Comment by iMartyn** [comment]
oh, gpg2 doesn't. gpgp1 gives
`sec::2048:1:AB4A6048CFAEF3F4:2014-01-10::::Martyn Ranyard (2014 onwards) <martyn@ranyard.info>:::`
`fpr:::::::::4C3101557BA583F47FF79776AB4A6048CFAEF3F4:`
`uid:::::::2391910FD2F9028AF3A6311C1DF800381ABDF214::keybase.io/imartyn <imartyn@keybase.io>:`
`ssb::2048:1:CBDE4D4D5DC2D226:2014-01-10:::::::`

**Comment by patrickxb** [comment]
Ok, so keybase is preferring to use `gpg2` if it finds it.  Perhaps the private key is only in the gpg1 keyring?

**Comment by maxtaco** [comment]
@iMartyn  which version of gpg2 do you have?

**Comment by iMartyn** [comment]
I didn't have gpg2 installed originally.
`gpg (GnuPG) 2.1.11`

**Comment by iMartyn** [comment]
wait... what.... ugh debian packaging is horrid.  gpgv2 is not gpg2

**Comment by iMartyn** [comment]
Okay, I can't seem to transfer the keys to gpg2 correctly.  I don't think it likes them not having a passphrase locally.  I guess this can be closed for now, with the slight issue that the error doesn't point out that the machine has gpg1 and gpg2 installed, which I'd consider a usability bug.

**Comment by maxtaco** [comment]
You can do `keybase config set gpg gpg` to disable defaulting to `gpg2`

**Comment by patrickxb** [comment]
Were you able to provision your linux machine successfully?

**Comment by iMartyn** [comment]
Yep, by doing the suggested command from @maxtaco, I'm doing that now and have got further.
